src/pobox.erl

```diff
@@ -221,13 +350,35 @@ handle_call(_From, _Msg, _StateName, _Data) ->
 handle_info({post, Msg}, StateName, State) ->
     %% We allow anonymous posting and redirect it to the internal form.
     ?MODULE:StateName(cast, {post, Msg}, State);
-
 handle_info(_Info, _StateName, _State) ->
     keep_state_and_data.
 
 %% @private
-terminate(_Reason, _StateName, _State) ->
-    ok.
+terminate(_Reason, _StateName, S=#state{owner=Owner, owner_pid=OwnerPid, heir=Heir, heir_data=HeirData, name=Name}) ->
+    case send_ownership_transfer(Owner, Heir, HeirData, Name) of
+        {ok, HeirPid} ->
+            erlang:unlink(OwnerPid), %% We keep track of the original owner we linked to and unlink from that in case the
+                                     %% name point to some new process.
+            erlang:link(HeirPid),
+            case Name of
+                Pid when is_pid(Pid) ->
+                    put('$ancestors', [HeirPid]),
```

Comment:
We should preserve other ancestors or take the ancestors of the heir (if available just switch the parent process head of the list otherwise). This is hacky but `$ancestors` hasn't changed in pretty much forever (I checked as far back as 2010 but I believe it was there as far back as 1996 when `proc_lib` was first added).

Still debating using monitors but it would be a less reliable but a cleaner and future-proof solution.